src/asciidoctor-extensions-api.js

```diff
@@ -9,6 +9,14 @@ var toBlock = function (block) {
   return block;
 };
 
+var registerExtension = function (registry, type, processor, name) {
+  if (typeof processor === 'function') {
```

Comment:
`processor` can be undefined if the user only used a "name". In this context "name" must be a "function", so we could do:

```js
if (typeof processor === 'undefined') {
  processor = name;
  name = null;
}
```

Arguably this is not a better solution :neutral_face: 

Also should we throw an exception in this case:

```js
this.block('not like this...'); // It ain't gonna work :|
```

Reply:
True. But I don't think it's the responsibility of the JavaScript API to validate the arguments. That should happen in the Ruby method.

The purpose of the API method is to prepare the arguments for the Ruby method. The only thing we have to do is coerce the function to a Ruby block (since Opal doesn't do this for us). Otherwise, we're just passing through the arguments as they were given.

Reply:
I've noticed that the Ruby side doesn't throw an intelligible error in this case. It probably should.

Reply:
Hmm, actually, it does. It just seems we aren't hitting it. I'll investigate.

Reply:
I'll adjust it so that we hit the correct error when only a string argument is passed in.

Reply:
Aha! I see. The error on the Ruby side only happens if the value is a symbol.

Reply:
As it turns out, this is the correct logic. The problem is that the exception in core is obtuse. I will address that for 1.5.7.

```
this.block('not like this...');
```

This is a valid call. What it means is to look for a class with the name "not like this...". Obviously, it will fail to resolve a class. But right now, core sends back the obtuse message "wrong constant name" instead of "Could not resolve class for name". That's what I'll focus on fixing.